docs/transport.rst

```diff
@@ -0,0 +1,214 @@
+    virtual bool recv_message(InputPacket& input_packet, int timeout) = 0;
+    virtual bool send_message(OutputPacket output_packet) = 0;
+    virtual int get_error() = 0;
+
+The first 4 virtual function are transport specific (Transport Layer).
+These functions are override by the ``UDPServerBase`` class which is in charge of managing the sessions between *Clients* and the *Agent*.
+
+On the other hand, the last 5 virtual function are platform specific (Platform Layer).
+These functions are override by the ``UDPServerLinux`` and ``UDPServerWindows`` for Linux and Windows systems respectively.
+
+Client Transport Architecture
+-----------------------------
+
+The *Client* transport architecture is analogous to the *Agent* architecture.
+There are also 3 different layers but instead of the Server Layer, there is a Session Layer.
+
+.. image:: images/client_transport_stack.svg
+
+* **Session Layer**: implements the XRCE protocol logic and it only knowns about sending and receiving messages.
+* **Transport Layer**: implements the sending and receiving **message functions** for each transport protocol, calling to the Platform Layer functions.
+  This layer only knows about sending and receiving messages through a given transport protocol.
+* **Platform Layer**: implements the sending and receiving **data functions** for each platform.
+  This layer only knows about sending and receiving raw data through a given transport in a given platform.
+
+UDP Transport Example
+^^^^^^^^^^^^^^^^^^^^^
+
+As an example, this subsection describes how the UDP transport is implemented in *eProsima Micro XRCE-DDS Client*.
+The figure below shows the *Client* transport architecture for the UDP transport.
+
+.. image:: images/client_transport_architecture.svg
+
+Similar to the *Agent* architecture, there is also an interface, ``uxrCommunication``, whose functions are call from the Session Layer.
+That is, each time a ``run_session`` is called, the Session Layer calls to ``send_msg_func`` and ``recv_msg_func`` without worrying about the transport protocol or the platform in use.
+This struct has the following virtual functions:
+
+.. code-block:: c
+
+    bool send_msg_func(void* instance, const uint8_t* buf, size_t len);
+    bool recv_msg_func(void* instance, uint8_t** buf, size_t* len, int timeout);
+    uint8_t comm_error_func(void);
+
+These functions are override by the ``uxrUDPTransport``, which is in charge of two main tasks:
+
+1. Override the communication interface functions. 
+   For example, in the case of the UDP protocol these functions are the followings:
+
+.. code-block:: c
+   
+    bool send_udp_msg(void* instance, const uint8_t* buf, size_t len);
+    bool recv_udp_msg(void* instance, uint8_t** buf, size_t* len, int timeout);
+    uint8_t get_udp_error(void);
+
+2. Offer to the user the initialization and close functions related with the transport protocol.
+   For example, in the case of the UDP protocol these functions are the followings:
```

Comment:
```suggestion
   For example, in the case of the UDP protocol these functions are the following:
```
I needed to search about it, the main rules are described here: https://forum.wordreference.com/threads/followings-or-following.865095/